sdk/storage/azure-storage-queue/azure/storage/queue/aio/_queue_client_async.py

```diff
@@ -94,14 +94,13 @@ def __init__(
         super(QueueClient, self).__init__(
             account_url, queue_name=queue_name, credential=credential, loop=loop, **kwargs
         )
-        self._client = AzureQueueStorage(self.url, base_url=self.url,
-                                         pipeline=self._pipeline, loop=loop)
-        self._client._config.version = get_api_version(kwargs)  # pylint: disable=protected-access
+        self._client_async = AzureQueueStorage(self.url, base_url=self.url, pipeline=self._pipeline, loop=loop)
```

Comment:
So I have slight concerns about this change and curious about your thoughts. I was running into a CI issue where the crux of the issue was that something was calling `self._client.__aenter__` (where self == QueueClient OR QueueServiceClient).

And so, in QueueServiceClient, this was working fine, since the `self._client` was correctly pointing to the async client.
However, for QueueClient, it was getting a non-async version of the client back. Upon further investigation, it would seem that the `super` call at 94 is to blame for populating the `self._client`, and then we doour initialization at line 97 for `self._client_async`.

So... consequently at line 100 here (end of `__init__`), we actually have both `self._client` and `self._client_async`:
![image](https://user-images.githubusercontent.com/101599632/235546944-24129aa8-9151-402b-8044-4c2a0e708547.png)

So I guess my questions are:

- Is this okay / expected? Seems sort of odd to have both clients accessible (but we only really want one)
- If not, how should we go about resolving this?